### DISCUSSION TITLE
Receiving different chain ID when connected to Opensea via WalletConnect

### DISCUSSION THREAD

**Discussion by aoifeinseoul** [WalletConnect/walletconnect-monorepo#4055] [Developer Support]
Hi, I am experiencing a similar issue as described in [this GitHub issue](https://github.com/WalletConnect/WalletConnectSwiftV2/issues/977).

When connected to OpenSea via WalletConnect, we encounter a problem where the wallet receives a transaction with a different chain ID than the actual chain ID that needs to be signed. For example, the following log shows our wallet attempting to buy a Klaytn NFT but receiving a transaction with a Polygon chain ID.

Can you look into this? Thanks!

iOS
SDK Version :  https://github.com/WalletConnect/WalletConnectSwiftV2/releases/tag/1.6.11 

🚄 [RelayClient]: Publishing payload on topic: 8515ae993be75ebdad6eff1852c9de31da75951d572b1c9abb8e89a58a6b5340 - 15:30:56.4650
🚄 [RelayClient]: received message: AJcw2eBgCPxJCqdkxfZ5j0/724A3MiDRmDeaiImMB9w7BAjLkN9K9Wrd7WXYtAch91xgixiDmb+nqbsSez1D9jC9JVA4cNok6O8WaNXnK6x0T6I2ADsm/3QfTCKwwZfGU3VGmx3E6a9xOwCR+6K4mMmW0ARMk0NXJJO6GNJysT31foAhDmrT4zsVFU1e4Xu+V9SjvHdHw1mSVCYPYDphoIwh/Z8k+BDOtZ+x1/7p5lgv3kZdIH6W7Gpey09/rSUDaXvCxl7IEVO1MQVJsGCp5nYTKH05Z6wkh/H9LB2X5NecanYMR6fx5F3uqiDaZMvChsq6I+PcsuLdKx2AGm7ORJzM3f+yIlGma7S9czrm9Sxd427yPK/WDBtdCjo0ocOgCY+MUGSSfSfTAJqcUxX5jPBxfwUdaX8VPtHxh2yDdXt5S7UkSgDPjvLJvXAok8qC4dXD7O91X6tDmVZABqbPoxQUqwqTZnUjKkERMJHPQHAKLCyjugqdaUn0haJREApMLE/yFhyg23tPdtZac/19VkP44nFagBfIsFfup1QoSDYyc4sAFmjYpiqcDLviCpLkhZ+biK6mB/2TWsSyfJNv2+ff4fRPNZ5trpHLD/h3G0OJQXbxFvlYgh3BqQN1oNC0z5iO/LT0imYdq7TsfI14sH1y2d0NNcCwlWxCnhVexELBv4fTkAkOSsO9BSFcWXkQdp5xB6GlsoCH2KSZ5Aj1J5bH3ybxCbZXkaFQO8qPAx0EQwlOZy3bQwABRjocWoHzhjEy7sB8zw5aKK8aSZOvZ98StqBM45EMmjaVhatcgHLsHYUcBWg0y9C9I7YDYJF0oWx4T76PWFnfUA/Ay/xuJyMMZdm6Cc1Ar9qU72JJcH29llgazg7nXxceL09IQh+40SFhcxXS1z31G8zPSiNlRCBlKspgYy3R3agimkBtAJE5pTpyFvOClzRfgNXhYDnmp9O0g+vU/BqTTvqpt9yy/ZhRJ9lAEYE4FMyacIebq20jzkfVmQ6tSKJLsK+ouCHbLnh2hgMyk7KRKsvlnpE5zJprQ6EzoYhdSsi+LTWCjk8CHaxTuyIPuCrYw9V1IW+elvObk+w41AuU9vtNkXjzXryfqtgkXPrhUOs5Vx9WnWiMyc2Ah/qgEXVHrI0I/zuIV/5PHGBJ/Yz+jz2EKzNVtk/9EMbAUvYlTXTTUAjl0O1MsHG3X/rUbMB4Y27iDMYCkCHXXtmOgDNrANGS+xN+KDLfS0Gde5X97Pu0Vj0cyaF8RMf5oWnrBMIzw97ll1tIBylD5vp9HFz40ZTz+LKw3Bgn0RAansNHIK7gsbWdWih09zpkeDOhq5CAClCWZ99i+2vcjUBb9gFaqqJuuqfFKGzCXlgbGFQQMYAdGeW6C0iWSMYFNKaAi8S42z9sKhK3ZbXuCMwvat5K5YM27g5DxE+byYWIxgAL1kGoR1u8rUSJtVlGifV6dY7K7V2f1ifcJ6yrnjQQdgN+yAQFqn6/EX2Ve770B8VkumEDQo+byk4vkTxooCnHuKj+iicfrMCdF0dNZhUG/H1tLvYXkJnHiImMGHozSEeJksjRbNt1Q08uikzB083+9JlJyVkwTJaEfyZ2QYDN4pSOKh80PCEq3ehaHlI+/J9fu7tcjKHa0rLjqH4cpPPe74xMHdGzJJCdfcddpJ0AGlbhUk1cN/6CUfVKFR4uFPtwfcui5p9oTh2ZJMF5Iz6RiAJF1A6fXUAMOL3wccLJLeIU5RZP8sfn+nrN/70B7EcPHjcMUi+q5JxiiXL+hB35BUeAXcLPr5AmpYGFVc//g1sooXuFRxOaAM0UeIFJSbpEKjw4Fg3FsExBxZiIMHM7ueie3TRh9WjFLx8FoVM+LGsatgbPTMsnMI6rAf9ecp1lcSQoLCuCuYroJ1hl+CcVrYCWCM/K62jnikEnbBXu+eGcHdok2k6wZq5t7STOViWOLDf9djIvchkd17QVxHonnD8JW3SxuQ0xC75QXhcUsiInf4CKi2YJ3xTiW9a9Vq6XqAUQvD9lU7CH6BNrspT70bYtWUD/Eb7jz+deuxSxSLEsXBzqw5KMZXVacY7qRCRVWomnWgUwYJ1nstPahEKGZ0w1T0TbFFPmbgc2BLRWFfKFy+luXhEYkTabXa4Y5ODsuLwJKuwPyRKBr3GjkjJMBJ9tLkLM9BQ93fj2WFlbTtAVv/yIFMGi/JeaBo9H5dy+h2wcKSAH0AWk1Ra2KFCncZst68kzI61hj5rYe1OBry5D9yszV5kMEgCv+xZDa0VOwefacR/WJXQ9z0LJ9FwTcQzFpZ5qWO/wN9cTRQTn166Wgwl6VUZEuMbhXfRnBE784bh22NfIc89924lTNTPxPJbkqQf9ZUOWpdCu5c/z6LV1JcogoTizK9gm4NeTR+My66jmyQXMefs1VcOGmLKAu+7F8Xear0rRnfinoD7oez0U3VvsaLQPkrAYzl50eqURwsXUUqZasFtgH638Pxt6Z51PAx0+yOSrkSfczi5i6Y/Df2JdJ1vHh8KaV548jgtW1B+Xjc42+1yij6fVnWRKtp8Pj69/+SKRKmCvbDP6yqO1P0BkU7lT4P83Q90VDrbyQJcFtD26FSBvqV1uQOKro8EhuVblSyTWyvCnGDlnhAvJs++ACIH/RD3VudVDe978YwJuwMMpqOb6Vtb0KKYc7fu8EYd1qIPVNzZDOrnVgoOJR4BQsHhuzZqSprhSJV1yNqar2io0kG2FoSWlbvZy7qBWvbM9MlEbkLNjWpnDJVTqt8vGugKIjVswemi08S6UCNplTcROc0f4BArgEoNKNvMJbsu+XCz+Y+CbXCNiQECwUppfgf3QjKoOG++uuCtdp3O7JJc3KWQEl03+sLUpt2ty3f1+6WR1OYyr2R4Is+e0hN4JBhcVmeAc174ukB9kZr8YVXbK36Kd7Z37vKa7/ZlRxPxTZCuJDCIpRtVWqCLspQT4ZApZfQO0a/SyqH+K+d7HNVERyLf++C+II8BxMRNK2wXIADIGb78E6H7sGAGInL0kvWSPc67BfbABK6t7DSCn9zkRHpVV+/oXLKCNK30y0ZJ8R+7S+/ACJzxU5/34Flu4DMiavB8e1RFH0bteLD7V/SqYm9QMXNruI/+a4zqmCbXP/BbcJvmvAyojeE7PRnJGvVLT9q8zGKI2u7mqm1kUFye9c87U567autVPJdOai23n5cgntN89M8k9c4A4TGNhLl+QWbgAcWmy1oRgU/T7nEzl8kVGYQdrSC0u+u9+UlVgNybqmfDrj2HFI48zL2tmmpekE8MM9SyKO5e1ZIDDiUo4SJhvdzL26r17W7bq/nEJPW9m8B4IBXliOxmqRnVlfyzg/twUsfv3yIG2zbo8Td78raGTPfiXj3NmzSMAt/2y7CgTdH9IFpTy8Wv1S9azrQ3+wvfhGejwZ7daBlYwynu81wiRBuHHjZqDi1nkotfwuOEsMrZcZzPNLVRW/u2cSu+UVXmIB3gxfYo0ZOmznEU9+S9DJnZsGPUZieJy6wJ2SNVoYnSMyNGYcnVd7csJ+sUYRL0gmX4NCXqeF+wX8umNbYFpwWwvucbqzll7RqZ0VpKZJvAv46hSt8cDwWUt9rlOQtdn5WL1sQE65cJznsv6ffa0oFB4W26bVO9uwVIcGLUwapxTvcme+ftAP4aFhWtBlSZIy93A3Q6eK37SEdqjAbT1iNq2PC/CnskPn/A3qT8rXcqbQw9HTS/GSDR3RwPlxUWA1g50ysQ4tAGNeD3+eH1PG6tuR0HSYhFVvd6SJGC14z0f6M+NJ2A5Tr0iHWRdirGMgi7L/Jbxz/JhDzrCkCoiuwgPhsLbrTmW3H7eBDXMWkC3+heIEgBgqQvoPpRMxmxsr2pEgfrOs7MZCpfwK1sJKKSnCkUiPNndFAJYuavYf6OvI1433qya0gC+dxt6bIHuocCfHjnU4gjvMs1Ahwr6DDM6p11nsIvRuq7aW8yhBF/kN4ebPKAINmPSRfgjclD/HLZtKU2+zlfLTsFeOGPSXLEZhuIVzszOaQVOlj+g/1/uke/wL72MIr203m7YKDqj+mXxSdyqEOC4V02jBysKRyPyNDf3EXNW2ySNuaCsn/hNZS/vinNkTe1aa96r3gjWXMCHOKMYl3PoaTLIOJTcM4dXUpjlHoItevt7zB2J4LshBIINijJvqyihEUcfMap/QDGKMVNPMTHYxd6ecjPDT22odXA0PSEQmNmHWKCvXlmiNhYrB5cUJ4QIRJ8uE1h7nGj46CtMJsPnSEPrMbuexGvgXVHhL0AbF4bQQXLH7668f+dJbyHm29yhiEb2OgU9ONDUVq4aOpPMb+pjfLUih6+Z28zGF3o+pqyDGr8eo2UFBpJVgzfBrc0f6KCaMIxFUlP1gBHORRZrVIqg+RtgwmI20i9LWrt7TtxHF3rxN975KX0/WxEU70LHgi3EhOF/TZ9yx0I9Hx645fQjIezTWSp/qQCQYt+S19CBUBYv6fGrh7JmPmLdJ0penD0gQ50/XuSlXoMYyVAtsRDk4maUMXk3NVae/4tEUVaASrutzLr5FJJEs4eIaRzixIZjaMwInXEiNztfqgJ3Elv8DUUMiM+KVK8VijmNtTfqSpE+15sOpuAFjRpY+snRNL2yY4Otdw4+sbiP77eKUiQnt4IEBRrktnzLnYMBkPW+mDw9ZcQLtxiOqRnYIdgG0r9Pn3JQgk4cnS8znvtCcD9+slr3DjBsyWccIa5DEKi0BU0f9JVCLAmtu9qyfMkFd2hWyg== on topic: 8515ae993be75ebdad6eff1852c9de31da75951d572b1c9abb8e89a58a6b5340 - 15:31:04.3870
======= session.request: Request(id: 1702449064311231, topic: "8515ae993be75ebdad6eff1852c9de31da75951d572b1c9abb8e89a58a6b5340", method: "eth_sendTransaction", params: AnyCodable: "[{"data":"0xe7acab24000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000006200000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000005200000000000000000000000000000000000000000000000000000000000000580000000000000000000000000bd1dfb0a32a632b0eb769d2998bce3bac682269d000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001600000000000000000000000000000000000000000000000000000000000000220000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000657942af0000000000000000000000000000000000000000000000000000000065a221150000000000000000000000000000000000000000000000000000000000000000360c6ebe000000000000000000000000000000000000000089140e2c60565e410000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000003000000000000000000000000d67b2bdd91ef23c4855b39f5ec9b3f6ad366eec70000000000000000000000000000000000000000000000000000000000001284000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004fefa17b724000000000000000000000000000000000000000000000000000004fefa17b7240000000000000000000000000000bd1dfb0a32a632b0eb769d2998bce3bac682269d000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002386f26fc10000000000000000000000000000000000000000000000000000002386f26fc100000000000000000000000000008de9c5a032463c561423387a9648c5c7bcc5bc90000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006a94d74f430000000000000000000000000000000000000000000000000000006a94d74f43000000000000000000000000000058e27d9d739e6f8537326a4c514b7ed4acc579f800000000000000000000000000000000000000000000000000000000000000404caa3cb262fa68549f1f51ae13b62eb1ffd6ed11c847136f7ecdc0da68f7bca284623226cb84587bdbbedf5a255552ddddd68f53327d369a31b1ba8595b222060000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000360c6ebe","from":”….","gas":"0x6800f","to":"0x00000000000000adc04c56bf30ac9d3c0aaf14dc","value":"0x8e1bc9bf040000"}]", chainId: eip155:137, expiry: nil)

```

**Comment by glitch-txs** [comment]
please bump the issue you linked, forwarding to the team

**Comment by aoifeinseoul** [comment]
@glitch-txs  the issue has been opened for 6months and is still not solved. Are you sure someone is looking into the issue? Would rather start a new discussion from my end if it is the case.
